src/log/file_appender.cpp

```diff
@@ -82,13 +82,12 @@ namespace fc {
                    out.flush();
                    out.close();
                }
+               string rez;
+               read_file_contents(link_filename, rez);
                remove_all(link_filename);  // on windows, you can't delete the link while the underlying file is opened for writing
-               if( fc::exists( log_filename ) )
-                  out.open( log_filename, std::ios_base::out | std::ios_base::app );
-               else
-                  out.open( log_filename, std::ios_base::out | std::ios_base::app);
-
+               out.open( log_filename, std::ios_base::out | std::ios_base::app );
                create_hard_link(log_filename, link_filename);
+               out << rez;
```

Comment:
I believe this new code is taking what is in "mylog.log" and copying it to "mylog.log.datetime" and leaving "mylog.log.datetime" open for writing.

That may be what we want for reboots, but is not what we want for normal rollovers. On a rollover, the file will continue to grow ever larger, as it has the previous rotation_interval messages in it as well.

My suggestion (I believe this is how it works in old log4j code):
- Always open mylog.log with the append flag. Always write log messages to mylog.log
- On a rollover, rename mylog.log to mylog.log.datetime, and create a new mylog.log, assigning it to "out" for future messages to write to it.

Depending on how the rename is done (not sure how boost implements this), this can prevent the windows problem, as a rename does not require the file to be closed. If any other processes are writing to it, they should be okay.

Or, if you would rather continue with the existing code, re-add the if statement and only copy the previous contents if fc::exists. This may be the safer option, as my solution mentioned above is dated and direct from (often faulty) memory chips in my head.

Reply:
where should I add if fc::exists statement and for which file to check ? clarify a bit more please ...

Thanks a lot !

Reply:
Don't take my word for it. Test to make sure what I'm saying is true. But perhaps something like:

``if( fc::exists( log_filename ) )``
``{``
&nbsp;&nbsp;&nbsp;``out.open( log_filename, std::ios_base::out | std::ios_base::app );``
&nbsp;&nbsp;&nbsp;``out << rez;``
``}``
``else``
&nbsp;&nbsp;&nbsp;``out.open( log_filename, std::ios_base::out | std::ios_base::app);``

Reply:
The ``if`` would need to replace the first out.open above. The pseudocode is:

- If (the file mylog.log.201806291800 exists) 
   - then open it and copy what was in mylog.log into it.
- Otherwise, open the file mylog.log.201806291800

Reply:
Perhaps it would be good to write some test cases. That way we know we're fixing the problem. Let me see if I can put one together quickly...

Reply:
@jmjatlanta Thanks a lot ! I'm checking too ;-)